docs/cookbook/evals/evaluating_web_search_agent.md

```diff
@@ -0,0 +1,347 @@
+                        )
+                self.search_history.append(query)
+            return "\n\n".join(parsed_results)
+
+        except Exception as e:
+            return f"{type(e)}: Failed to search the web for text"
+
+    @openai.call(model="gpt-4o-mini", stream=True)
+    @prompt_template(
+        """
+        SYSTEM:
+        You are an expert web searcher. Your task is to answer the user's question using the provided tools.
+        The current date is {current_date}.
+
+        You have access to the following tools:
+        - `_web_search`: Search the web when the user asks a question. Follow these steps for EVERY web search query:
+
+            1. There is a previous search context: {self.search_history}
+            2. There is the current user query: {question}
+            3. Given the previous search context, generate multiple search queries that explores whether the new query might be related to or connected with the context of the current user query. 
+                Even if the connection isn't immediately clear, consider how they might be related.
+        - `extract_content`: Parse the content of a webpage.
+
+        When calling the `web_search` tool, the `body` is simply the body of the search
+        result. You MUST then call the `parse_webpage` tool to get the actual content
+        of the webpage. It is up to you to determine which search results to parse.
+
+        Once you have gathered all of the information you need, generate a writeup that
+        strikes the right balance between brevity and completeness based on the context of the user's query.
+
+        MESSAGES: {self.messages}
+        USER: {question}
+        """
+    )
+    async def _call(self, question: str) -> openai.OpenAIDynamicConfig:
+        return {
+            "tools": [self._web_search, extract_content],
+            "computed_fields": {
+                "current_date": datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+            },
+        }
+
+    async def _step(self, question: str):
+        response = await self._call(question)
+        tools_and_outputs = []
+        async for chunk, tool in response:
+            if tool:
+                tools_and_outputs.append((tool, tool.call()))
+            else:
+                print(chunk.content, end="", flush=True)
+        if response.user_message_param:
+            self.messages.append(response.user_message_param)
+        self.messages.append(response.message_param)
+        if tools_and_outputs:
+            self.messages += response.tool_message_params(tools_and_outputs)
+            await self._step("")
+
+    async def run(self):
+        while True:
+            question = input("(User): ")
+            if question == "exit":
+                break
+            print("(Assistant): ", end="", flush=True)
+            await self._step(question)
+            print()
+```
+
+For more information regarding this code snippet, refer to the Web Search Agent cookbook.
```

Comment:
nit: I would remove this line in favor of merging the intended point with the first line, which should be part of the collapsible block